src/actions/post_build.rs

```diff
@@ -81,7 +84,71 @@ impl PostBuildHandler {
                 self.notifier.notify_end_diagnostics();
                 self.active_build_count.fetch_sub(1, Ordering::SeqCst);
             }
+            BuildResult::CargoError {
+                error,
+                stdout,
+                manifest_path,
+            } => {
+                trace!("build - CargoError: {}, stdout: {:?}", error, stdout);
+                self.notifier.notify_begin_diagnostics();
+
+                if let Some(manifest) = manifest_path {
+                    // if possible generate manifest diagnostics instead of showMessage
+                    self.handle_cargo_error(manifest, &error, &stdout);
+                } else if self.shown_cargo_error.swap(true, Ordering::SeqCst) {
+                    warn!("Not reporting: {} {:?}", error, stdout);
+                } else {
+                    let stdout_msg = if stdout.is_empty() {
+                        "".to_string()
```

Comment:
why we cannot simply return `stdout`?

I'm not an immediate rustican, but that is how I would solve it.